spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -131,12 +133,12 @@ public SparkTable(Table icebergTable, boolean refreshEagerly) {
   public SparkTable(Table icebergTable, String branch, boolean refreshEagerly) {
     this(icebergTable, refreshEagerly);
     this.branch = branch;
+    final Snapshot snapshot = icebergTable.snapshot(branch);
     ValidationException.check(
-        branch == null
-            || SnapshotRef.MAIN_BRANCH.equals(branch)
-            || icebergTable.snapshot(branch) != null,
+        branch == null || SnapshotRef.MAIN_BRANCH.equals(branch) || snapshot != null,
         "Cannot use branch (does not exist): %s",
         branch);
+    this.snapshotId = snapshot.snapshotId();
```

Comment:
Won't this throw an NPE in some cases as `snapshot` could be null?

Also, I am not sure this logic is correct. Tables loaded for a particular snapshot ID and for a particular branch may not be logically equal, more operations could happen to the branch upon its initial load.

Reply:
it looks like some tests fail with a NPE at exactly this line

Reply:
Sorry, I misunderstood the constructor. Fixed and confirmed on my local. `./gradlew -DsparkVersions=3.5 -DscalaVersion=2.12 -DhiveVersions= -DflinkVersions= :iceberg-spark:iceberg-spark-3.5_2.12:check :iceberg-spark:iceberg-spark-extensions-3.5_2.12:check :iceberg-spark:iceberg-spark-runtime-3.5_2.12:check -Pquick=true -x javadoc`